drink/src/session.rs

```diff
@@ -395,3 +470,71 @@ impl<R: Runtime> Session<R> {
         self.sandbox.override_debug_handle(d);
     }
 }
+
+/// A struct representing the result of parsing a `.contract` bundle file.
+///
+/// It can be used with the following methods of the `Session` struct:
+/// - `deploy_bundle`
+/// - `deploy_bundle_and`
+/// - `upload_bundle`
+/// - `upload_bundle_and`
+pub struct ContractBundle {
+    /// WASM blob of the contract
+    bytes: Vec<u8>,
+    /// Transcoder derived from the ABI/metadata
+    transcoder: Rc<ContractMessageTranscoder>,
+}
+
+impl ContractBundle {
+    /// Load and parse the information in a `.contract` bundle under `path`, producing a `ContractBundle` struct.
+    pub fn load<P>(path: P) -> Result<Self, SessionError>
+    where
+        P: AsRef<std::path::Path>,
+    {
+        let metadata: ContractMetadata = ContractMetadata::load(&path).map_err(|e| {
+            SessionError::BundleLoadFailed(format!("Failed to load the contract file:\n{e:?}"))
+        })?;
+
+        let ink_metadata = serde_json::from_value(serde_json::Value::Object(metadata.abi))
+            .map_err(|e| {
+                SessionError::BundleLoadFailed(format!(
+                    "Failed to parse metadata from the contract file:\n{e:?}"
+                ))
+            })?;
+
+        let bytes = metadata
```

Comment:
awful nit, but I would replace `bytes` and `ink_metadata` definition order (so that things are epsilon more local) :P